components/eamxx/src/share/io/eamxx_io_utils.cpp

```diff
@@ -129,21 +129,25 @@ create_diagnostic (const std::string& diag_field_name,
   //       of each group in the matches output var!
   // Note: use raw string syntax R"(<string>)" to avoid having to escape the \ character
   // Note: the number for field_at_p/h can match positive integer/floating-point numbers
-  std::regex field_at_l (R"(([A-Za-z0-9_.+-]+)_at_(lev_(\d+)|model_(top|bot))$)");
-  std::regex field_at_p (R"(([A-Za-z0-9_.+-]+)_at_(\d+(\.\d+)?)(hPa|mb|Pa)$)");
-  std::regex field_at_h (R"(([A-Za-z0-9_.+-]+)_at_(\d+(\.\d+)?)(m)_above_(sealevel|surface)$)");
+  // Start with a generic for a field name allowing for all letters, all numbers, dash, dot, plus, minus, product, and division
+  // Escaping all the special ones just in case
+  std::string generic_field = "([A-Za-z0-9_.+\\-\\*\\÷]+)"; 
+  std::regex field_at_l (R"()" + generic_field + R"(_at_(lev_(\d+)|model_(top|bot))$)");
```

Comment:
What's the leading "()" for?

Reply:
std::regex constructor takes a string, and generic_field is already a std::string (otherwise the + operator would not work). The reason why we need the R in other  strings is to prevent stuff like `\\` to be escaped. And speaking of, you can just do

```c++
std::string generic_field = "([A-Za-z0-9_.+-*÷]+)"; 
```
no need to escape characaters inside `[]`. And since you don't have fancy chars, no need for R. Btw, why `÷` instead of `/`? It's not a standard char, so maybe not what ppl would use...

Reply:
Uhm, `÷` is not ASCII though, which may be undesirable...